src/map/mob.c

```diff
@@ -3860,6 +3900,192 @@ static inline int mob_parse_dbrow_cap_value(int class_, int min, int max, int va
 	return value;
 }
 
+/**
+ * Reads the settings for one random option slot of a random option drop group.
+ * @param optslot : The slot entry from config file
+ * @param n : slot index
+ * @param group_id : Group index
+ * @param group : group name (used in messages)
+ * @return true if it succesfully read, false otherwise
+ */
+static bool mob_read_optdrops_optslot(struct config_setting_t *optslot, int n, int group_id, const char *group)
+{
+	nullpo_retr(false, optslot);
+	nullpo_retr(false, group);
+
+	// Structure:
+	//	{
+	//		Rate: chance of option 1 (int)
+	//		OptionName1: value
+	//		OptionName2: [min, max]
+	//		OptionName3: [min, max, rate]
+	//      ....
+	//	}
+
+	int drop_rate; // The rate for this option to be dropped (Rate field)
+	if (!libconfig->setting_lookup_int(optslot, "Rate", &drop_rate)) {
+		ShowWarning("mob_read_optdrops_optslot: Missing option %d rate in group %s, skipping.\n", n + 1, group);
+		return false;
+	}
+
+	int count = libconfig->setting_length(optslot);
+	if (count <= 1) { // 1 = Rate
+		ShowWarning("mob_read_optdrops_optslot: Option %d of %s group doesn't contain any possible options, skipping.\n", n + 1, group);
+		return false;
+	}
+
+	struct optdrop_group_optslot *entry = &(mob->opt_drop_groups[group_id].optslot[n]);
+	entry->options = aCalloc(sizeof(struct optdrop_group_option), count);
+	
+	int idx = 0,
+		i = 0;
+	struct config_setting_t *opt = NULL;
+	bool calc_rate = false;
+	while (i < count && (opt = libconfig->setting_get_elem(optslot, i))) {
+		++i;
```

Comment:
better extract code from inside while into separate function

Reply:
like that?

Reply:
while in old function, but all inside while in new one. (probably ++i still in old function)
this need for simpler code and for plugins to allow change something. for example change only reading min, max fields or add custom fields here and without changing anything else.

Reply:
I think it is correct now